tests/test_helpers.py

```diff
@@ -183,17 +183,28 @@ def test__get_image_extra_build_command_options(git_repo):
         ("1.2.3-0.dev", "2.0.0", 0, "1.2.3-0.dev"),
         # ignore invalid semver tags
         ("1.2.3-0.dev", "x.y.z", 10, "1.2.3-0.dev"),
+        # autoincrement latest tag
+        ("major", "1.2.3", 10, "2.0.0-0.dev"),
+        ("minor", "1.2.3", 10, "1.3.0-0.dev"),
+        ("patch", "1.2.3", 10, "1.2.4-0.dev"),
```

Comment:
What is the intended behavior if the latest tag is a pre-release like `1.2.3-beta.1`?

I'm thinking the point of autoincrementing is to avoid the incorrect ordering by appending `-0.dev`, but appending more characters to a pre-release like `1.2.3-beta.1` will not lead to incorrect ordering. So, we don't auto-increment the latest tag if its `1.2.3-beta.1` right?

Reply:
Looking at the current implementation, I understand this test case to pass:

```
("patch", "1.2.3-beta.1", 10, "1.2.4-0.dev"), # but I expected `1.2.3-beta.1` here
```